arviz/sel_utils.py

```diff
@@ -0,0 +1,224 @@
+"""Utilities for selecting and iterating on xarray objects."""
+from itertools import product, tee
+
+import numpy as np
+import xarray as xr
+
+
+def selection_to_string(selection):
+    """Convert dictionary of coordinates to a string for labels.
+
+    Parameters
+    ----------
+    selection : dict[Any] -> Any
+
+    Returns
+    -------
+    str
+        key1: value1, key2: value2, ...
+    """
+    return ", ".join(["{}".format(v) for _, v in selection.items()])
+
+
+def make_label(var_name, selection, position="below"):
+    """Consistent labelling for plots.
+
+    Parameters
+    ----------
+    var_name : str
+       Name of the variable
+
+    selection : dict[Any] -> Any
+        Coordinates of the variable
+    position : str
+        Whether to position the coordinates' label "below" (default) or "beside"
+        the name of the variable
+
+    Returns
+    -------
+    label
+        A text representation of the label
+    """
+    if selection:
+        sel = selection_to_string(selection)
```

Comment:
just thinking about this

Let's say we have `theta[3]` with coords ["Ferrari", "Fiat", "Ford"]

Could we create labels as (can index origin work with these?)

    "theta[0] Ferrari"
    "theta[1] Fiat"
    "theta[2] Ford"

or 

    "theta[0]\nFerrari"
    "theta[1]\nFiat"
    "theta[2]\nFord"

For this to happen, do we need to update selection (`xarray_sel_iter`?). And will index_origin work when coordinates are explicitly set?

Reply:
I want this to be possible and should be if we support custom labeling function, I am not sure what would be a good way though. 

I do think we will have to update xarray_sel_iter or use xarray_sel_iter to create a function targeted to labeling. In addition to the index, I also would like to expose the dimension so that `theta\nbrand: Ferrari`, `theta\nbrand: Fiat` are also possible. It may also be interesting to keep coordinate values simple (maybe abbreviations) and then have some attribute `label` (and `labels` attr in coordinates) so that users could select variable `theta` using `theta` but set its name to `$\theta$` for plots to show the greek letter. This gets crazy quite fast though. Eventually, `var_name`, `dim`/`coord_name`, `coord_value`, `coord_idx` and maybe `var_label`, `dim_label` and `coord_label` would be exposed when creating labels so custom functions can play with them to get the desired labels.

Regarding `coord_idx`, I think it should not include `index_origin` to get the same value that would be used by `isel`, but we should have an example on adding `index_origin` to `coord_idx` inside the labeller function.